File: TikraqApp/src/app/model/Curso.ts
import {Lengua} from './Lengua';
import {Usuario} from './Usuario';
import {Leccion} from './Leccion';

export class Curso {
  id: number;
  nombreCurso: string;
  descripcion: string;
  lenguasLenguaid: { id:number };
  usuariosUsuarioid: { id: number };
  lecciones: Leccion[] = [];

  constructor(id: number, nombreCurso: string, descripcion: string, lenguasLenguaid: { id:number }, usuariosUsuarioid: { id: number }) {
    this.id = id;
    this.nombreCurso = nombreCurso;
    this.descripcion = descripcion;
    this.lenguasLenguaid = lenguasLenguaid;
    this.usuariosUsuarioid = usuariosUsuarioid;
  }

}

File: src/main/java/com/example/tikraq/entities/Curso.java
package com.example.tikraq.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "cursos")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Curso {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cursoid", nullable = false)
    private Integer id;

    private String nombreCurso;

    private String descripcion;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "lenguas_lenguaid")
    private Lengua lenguasLenguaid;


    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JsonIgnore
    @JoinColumn(name = "usuarios_usuarioid")
    private Usuario usuariosUsuarioid;

    @OneToMany(mappedBy = "cursosCursoid", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Leccion> lecciones = new LinkedHashSet<>();

}